### ISSUE TITLE
Label component does not support htmlFor prop

### ISSUE THREAD

**Issue by nicholasrice** [Microsoft/fast-dna#1317]
Attempting to add an `htmlFor` prop to the Label component causes a TypeScript compile error.

Expected: Label should accept an `htmlFor` prop as an `unhandledProp` without causing a TypeScript error.

**Comment by chrisdholt** [comment]
@nicholasrice I think the issue here *may* be that `for` is not a valid attribute on legend which is the alternative tag for this. 

I would propose either:

1. Enumerate `htmlFor` as a handled prop and only apply if tag is label.
2. Remove legend as an optional tag for label and accept as `htmlFor` as an unhandledProp

**Comment by nicholasrice** [comment]
Yea I think the prop interface is just wrong - I think it should be the following:

```
export interface LabelUnhandledProps extends React.LabelHTMLAttributes<HTMLLabelElement | HTMLLegendElement>, React.HTMLAttributes<HTMLLabelElement | HTMLLegendElement> {}
```

**Comment by nicholasrice** [comment]
closed by #1417
